src/app/components_ngrx/work-item-detail/work-item-detail.component.html

```diff
@@ -155,6 +66,96 @@
           [class.col-lg-12]="detailContext=='preview'">
           <!-- assignee -->
           <div class="form-group">
+            <!-- work-item-type dropdown -->
+            <section class="f8-detail__area">
+              <label class="pull-left padding-left-15 control-label">
+                Type
+              </label>
+              <div class="f8-detail__content">
+                <span *ngIf="loadingTypes"
+                  class="spinner spinner-sm spinner-inline">
+                </span>
+                <span class="dropdown" dropdown
+                  *ngIf="loggedIn && !loadingTypes" 
+                  [isDisabled]="!workItem?.editable">
+                  <span class="pointer"
+                    id="wi-type"
+                    dropdownToggle>
+                    <i class="dropdown-icon"
+                      [ngClass]="type ?
+                      type.icon : '' ">
+                    </i>
+                    <span class="padding-left-5">
+                      {{ type ? type.name : '' }}
+                    </span>
+                  </span>
+                  <ul class="dropdown-menu"
+                    id="wi-type-dropdown"
+                    role="menu"
+                    *dropdownMenu>
+                    <li *ngFor="let type of wiTypes"
```

Comment:
While using `common-selector`
wiTypes should come directly from `workItemTypeSource` i.e. `workItemTypesource | async`